qos/cosmos/check_cometbft_health.go

```diff
@@ -7,14 +7,19 @@ import (
 	"github.com/buildwithgrove/path/qos/jsonrpc"
 )
 
+// TODO_IMPROVE: Create an interface that partially or completely captures the
```

Comment:
IMO this is not necessary. An early iteration of the QoS package (for EVM) experimented with a `check` interface for all endpoint checks and was eventually discarded (after review by Arash) in favour of concrete structs per check.

The reasoning is that individual checks vary enough in implementation that the concrete struct better serves the needed functionality. Additionally, since all service-level QoS logic is service specific by design and is never exported outside the package, an interface does not add much advantage.